projects/project_ideas.md

```diff
@@ -0,0 +1,59 @@
+# CS 7290: Project Ideas
+**Prof. Robert Ness**
+
+## 1. State-of-the-art Causal Inference methods
+
+You are encouraged to explore Causal Inference literature and to explore the papers that suits your research interests. For reference, take a look at the proceedings of NeurIPS and UAI for last four years. The applications of these methods do not limit to AI and computer science. They expand to Epidemiology, Social sciences, Medicines, Statistics and Political sciences. You can pick the case studies that suit your interest.
+
+**Expectations**: Comprehensive literature survey and understanding of the problem, implementation of state-of-the-art methods, comparative analysis of similar methods and suggesting possible improvements.
+
+
+## 2. Causal Inference with Probabilistc Generative Models
+
+[https://arxiv.org/abs/1805.06826]()
+
+**Introduction**:
+
+Causal inference from observational data often assumes “ignorability,” that all confounders are observed. This assumption is standard yet untestable. However, many scientific studies involve multiple causes, different variables whose effects are simultaneously of interest. *The deconfounder*, an algorithm that combines unsupervised machine learning and predictive model checking to perform causal inference in multiple-cause settings. *The deconfounder* infers a latent variable as a substitute for unobserved confounders and then uses that substitute to perform causal inference. The deconfounder provides a checkable approach to estimating closer-to-truth causal effects.
+
+**Keywords**: Potential Outcomes, Probabilistic Generative Models.
+
+**Data**: Any dataset that comes with structured content and response. For example, Actor and movie revenue or Smoking cancer dataset (from paper), Topics (extracted from documents) and response(for example, # views of document).
+
+**Expectations**: Deconfounded causal effect. For example, What is
```

Comment:
Let's keep this simple and specific.

"
* Implement the [TMDB 5000 Movie Dataset](https://www.kaggle.com/tmdb/tmdb-movie-metadata) case study described in the paper
* Implement the paper in Pyro, making use of the do-operator.
* Compare the intervention model to a regular non-causal predictive model.  Find examples where the predictive model gets it wrong "eg predicting high box office revenue when it stars the actress who plays Ms. Money in James Bond films is cast".  Show that the causal model provides a more sensible prediction.
* Demonstrate a counterfactual ("How much more would the film had made if Brad Pitt been cast instead of Nicholos Cage)
* (Bonus) Expand the model to include other causes of revenue such as genre, or use a tensor representation of actor combinations.
* Submission should be a Github repository with code and Jupyter notebooks that import that code and demonstrate predictive queries.  These notebooks should be used to give live examples during the class presentation.  Code should be clean and well-documented or points will be deducted.
* If you would like to use another different data set from the paper, no problem.  If you want to use a data set not in the paper, or a different framework other than Pyro, talk to the instructor or the TA's.
"